flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/CepOperator.java

```diff
@@ -328,7 +328,11 @@ public void onEventTime(InternalTimer<KEY, VoidNamespace> timer) throws Exceptio
         // STEP 4
         updateNFA(nfaState);
 
-        if (!sortedTimestamps.isEmpty() || !partialMatches.isEmpty()) {
+        if (!sortedTimestamps.isEmpty()
+                || !partialMatches.isEmpty()
+                // register timer if there are counters kept, so that we can clear them up
+                // if time advances without new events
+                || partialMatches.getEventCounters().hasNext()) {
```

Comment:
Hello, is there any possibility of not registering the timer here?
I may have found a performance problem caused by registering the timer, which may have something to do with the code in this place. Can you take a look at  this issue if possible?  [](https://issues.apache.org/jira/browse/FLINK-23890)